include/circt/Dialect/HW/HWOps.h

```diff
@@ -199,10 +199,13 @@ class HWModulePortAccessor {
   // Returns the i'th/named input port of the module.
   Value getInput(unsigned i);
   Value getInput(StringRef name);
+  ValueRange getInputs();
 
   // Assigns the i'th/named output port of the module.
   void setOutput(unsigned i, Value v);
   void setOutput(StringRef name, Value v);
+
+  const ModulePortInfo &getInfo();
```

Comment:
getInfo is a pretty general name, maybe call it getModulePortInfo like the other function in this file, or getPorts like the method on HWModuleOp?